examples/binarySearch/binarySearch.act

```diff
@@ -0,0 +1,38 @@
+behaviour init of BinarySearch
+interface constructor()
+
+creates
+	uint256[] array := []
+
+invariants
+	sorted(array)
+
+where
+
+sorted(uint256[] a) = 
+	forall (uint256 i) i > 0 && i < array.length => array[i] >= array[i - 1] 
+
+
+// Returns the index of elem if elem in array,
+// otherwise returns -1.
+behaviour search of BinarySearch
+interface search(uint256 elem)
+
+returns binarySearch(array, elem, 0, array.length)
+
+// postcondition
+// if elem in array, i | array[i] == elem
+// if elem not in array, -1
```

Comment:
@MrChico what do you think about this /\?

Reply:
I would write it like this, I think:
```
ensures

if elem in array then binarySearch(array, elem, 0, array.length) == elem else binarySearch(array, elem, 0, array.length) == -1
```
```

Reply:
or, if you want to refer to `i` here, add `i := binarySeach(array, elem, 0, array.length)` to the where section

Reply:
ok, so we're going with `e in c` for `e is a member of container c`? Did we have that syntax already?

Reply:
I don't think we introduce it yet but I like it!

Reply:
updated